src/automatic-annotations/shared/column-renderers/TaxonomicScope.tsx

```diff
@@ -0,0 +1,58 @@
+import { ExpandableList } from 'franklin-sites';
+
+import TaxonomyView from '../../../shared/components/entry/TaxonomyView';
+
+import { ARBAAPIModel } from '../../arba/adapters/arbaConverter';
+import { UniRuleAPIModel } from '../../unirule/adapters/uniRuleConverter';
+import { TaxonomyDatum } from '../../../supporting-data/taxonomy/adapters/taxonomyConverter';
+
+import helper from '../../../shared/styles/helper.module.scss';
+
+type CustomTaxonScope = {
+  negative: boolean;
+  taxon: TaxonomyDatum;
+};
+
+const TaxonomicScope = ({
+  mainRule,
+}: Partial<UniRuleAPIModel | ARBAAPIModel>) => {
+  // Using a map in order to remove duplicates
+  const taxonScopeMap = new Map<number, CustomTaxonScope>();
+  for (const conditionSet of mainRule?.conditionSets || []) {
+    for (const condition of conditionSet.conditions || []) {
+      if (condition.type !== 'taxon') {
+        continue; // eslint-disable-line no-continue
+      }
+      for (const conditionValue of condition.conditionValues || []) {
+        // This shouldn't happen
+        /* istanbul ignore if */
+        if (!conditionValue.cvId) {
+          continue; // eslint-disable-line no-continue
+        }
+        const taxonId = +conditionValue.cvId;
+        taxonScopeMap.set(taxonId, {
```

Comment:
Just an idea, check if `taxonId` is already present and if so skip it to avoid rewriting (presumably) the same thing? Unless this would mess up the order?

Reply:
I added a check to avoid re-setting a new object if already seen this taxon. Again, not sure if that would ever happen, or even make sense in the context of the rule, but oh well...

Reply:
I've just consoled this and it does happen though admittedly not enough (~3% of rows in table view) to make a difference.

Reply:
I'll have a look then, to try to understand what it means and if it's just some edge-cases or something we need to tackle more specifically.